metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -262,14 +263,15 @@ func getResourceMetadataWatchers(config *kubernetesConfig, resource kubernetes.R
 
 	options := kubernetes.WatchOptions{
 		SyncTimeout: config.SyncPeriod,
+		Namespace:   config.Namespace,
```

Comment:
Not sure if it really matters but I'm adding the check anyways. I would expect an empty value to have no impact.

Reply:
interesting why this check is added only for `options` that are used for `nodeWatcher`, but not for `watcher` (in autodiscover - https://github.com/elastic/beats/blob/fb2f53c4677b1677fbd572148399bcb693ab11d1/libbeat/autodiscover/providers/kubernetes/pod.go#L98-L100), maybe it is just a redundancy there.  Anyway please let me know in case you figure it out. I am fine to keep as it is now

Reply:
let me try to remove it and see what happens.